### PR TITLE
home-assistant-custom-components.somweb: init at 1.1.0

### PR DESCRIPTION
**Pull request by uvNikita**
## Description of changes

- home-assistant-custom-components.somweb: init at 1.1.0
- python3Packages.somweb: init at 1.2.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/servers/home-assistant/custom-components/somweb/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  propagatedBuildInputs = [
+    somweb
+  ];
+
+  meta = with lib; {
+    changelog = "https://github.com/taarskog/home-assistant-component-somweb/releases/tag/v${version}";
+    description = "Custom component for Home Assistant to manage garage doors and gates by Sommer through SOMweb";
+    homepage = "https://github.com/taarskog/home-assistant-component-somweb";
+    maintainers = with maintainers; [ uvnikita ];
+    license = licenses.mit;
```
[the remark this answers is not in the document]
Comment (uvNikita) [reply pkgs/servers/home-assistant/custom-components/somweb/default.nix:28]: No license file in this repo, should I remove it?

[the remark this answers is not in the document]
Comment (mweinelt) [reply pkgs/servers/home-assistant/custom-components/somweb/default.nix:28]: Means it is covered by copyright and therefore unfree. Probably best to file an issue.

**Comment by uvNikita** [comment]
Rebased:

- added `mainProgram`
- added `doCheck = false`
- reformatted `pkgs/servers/home-assistant/custom-components/somweb/default.nix` file

Also created issue in the component repository to clarify the license.

File: pkgs/servers/home-assistant/custom-components/somweb/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = "home-assistant-component-somweb";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-anOcpaGeblFVaP2EFVuxx1EuXnNgxy/QoYqvYJMv1Fo=";
+  };
+
+  propagatedBuildInputs = [ somweb ];
```
Comment (dotlambda) [inline pkgs/servers/home-assistant/custom-components/somweb/default.nix:20]:
```suggestion
  dependencies = [ somweb ];
```

**Comment by uvNikita** [comment]
@mweinelt MIT license was added to the repository 🥳

**Comment by uvNikita** [comment]
Rebased:
- changed from `propagatedBuildInputs`  to `dependencies`.

I think this should be ready now.

**Review by mweinelt — APPROVED** [review]

**Pull request merged**